src/ADC.c

```diff
@@ -0,0 +1,374 @@
+		ADC_RegularChannelConfig(ADC1, ADC_Channel_12, 1, ADC_SampleTime_56Cycles); //PC2
+	}
+	else if(ADC_sensor_x == Water_ADC){
+		ADC_RegularChannelConfig(ADC1, ADC_Channel_11, 1, ADC_SampleTime_56Cycles); //PC1
+	}
+	//Enable ADC
+	ADC1->CR2 |= ADC_CR2_ADON;
+}
+
+static uint16_t double_to_int(double double_x){
+	uint16_t double_int = (uint16_t)double_x;
+	if((double_x - (double)double_int) >= 0.5) double_int++;
+	return double_int;
+}
+
+static double calculate_average(uint8_t Read_Position_x){
+	average = 0;
+	for(ADC_count = 0; ADC_count < 5; ADC_count++){
+	    average += (double)ADC_Values[Read_Position_x][ADC_count];
+	 }
+	average /= 5;
+	return average;
+}
+
+extern void set_motor_current_temp_MUX(ADC_sensors_t ADC_sensor_x) {
+	switch (ADC_sensor_x) {
+		case Curr_ADC1:
+			GPIOC->ODR |= GPIO_Pin_9; //turn on PC9
+			GPIOC->ODR &= ~(GPIO_Pin_7 | GPIO_Pin_8);
+			break;
+		case Curr_ADC2:
+			GPIOC->ODR |= GPIO_Pin_9 | GPIO_Pin_8; //turn on PC9 and PC8
+			GPIOC->ODR &= ~(GPIO_Pin_7);
+			break;
+		case Curr_ADC3:
+			GPIOC->ODR |= GPIO_Pin_9 | GPIO_Pin_8 | GPIO_Pin_7; //turn on PC9, PC8, and PC7
+			break;
+		case Curr_ADC4:
+			GPIOC->ODR |= GPIO_Pin_9 | GPIO_Pin_7; //turn on PC9 and PC7
+			GPIOC->ODR &= ~(GPIO_Pin_8);
+			break;
+		case Curr_ADC5:
+			GPIOD->ODR |= GPIO_Pin_9; //turn on PD9
+			GPIOD->ODR &= ~(GPIO_Pin_8);
+			GPIOB->ODR &= ~(GPIO_Pin_15);
+			break;
+		case Curr_ADC6:
+			GPIOD->ODR |= GPIO_Pin_9 | GPIO_Pin_8; //turn on PD9 and PD8
+			GPIOB->ODR &= ~(GPIO_Pin_15);
+			break;
+		case Curr_ADC7:
+			GPIOD->ODR |= GPIO_Pin_9 | GPIO_Pin_8; //turn on PD9 and PD8
+			GPIOB->ODR |= GPIO_Pin_15; //turn on PB15
+			break;
+		case Curr_ADC8:
+			GPIOD->ODR |= GPIO_Pin_9; //turn on PD9
+			GPIOB->ODR |= GPIO_Pin_15; //turn on PB15
+			GPIOD->ODR &= ~(GPIO_Pin_8);
+			break;
+		case Temp_ADC1:
+			GPIOC->ODR |= GPIO_Pin_8; //turn on PC8
+			GPIOC->ODR &= ~(GPIO_Pin_9 | GPIO_Pin_7);
+			break;
+		case Temp_ADC2:
+			GPIOC->ODR |= GPIO_Pin_7; //turn on PC7
+			GPIOC->ODR &= ~(GPIO_Pin_8 | GPIO_Pin_9);
+			break;
+		case Temp_ADC3:
+			GPIOC->ODR &= ~(GPIO_Pin_9 | GPIO_Pin_8 | GPIO_Pin_7); //turn off PC7-PC9
+			break;
+		case Temp_ADC4:
+			GPIOC->ODR |= GPIO_Pin_8 | GPIO_Pin_7; //turn on PC7-PC8
+			GPIOC->ODR &= ~(GPIO_Pin_9);
+			break;
+		case Temp_ADC5:
+			GPIOD->ODR |= GPIO_Pin_8; //turn on PD8
+			GPIOD->ODR &= ~(GPIO_Pin_9);
+			GPIOB->ODR &= ~(GPIO_Pin_15);
+			break;
+		case Temp_ADC6:
+			GPIOB->ODR |= GPIO_Pin_15; //turn on PB15
+			GPIOD->ODR &= ~(GPIO_Pin_8 | GPIO_Pin_9);
+			break;
+		case Temp_ADC7:
+			GPIOD->ODR &= ~(GPIO_Pin_8 | GPIO_Pin_9); //turn off PD8-PD9
+			GPIOB->ODR &= ~(GPIO_Pin_15); //turn off PB15
+			break;
+		case Temp_ADC8:
+			GPIOD->ODR |= GPIO_Pin_8; //turn on PD8
+			GPIOB->ODR |= GPIO_Pin_15; //turn on PB15
+			GPIOD->ODR &= ~(GPIO_Pin_9);
+			break;
+		case Water_ADC:
+			break;
+		default:
+			break;
+	}
+}
+
+extern void read_ADC(ADC_sensors_t ADC_sensor_x){
+
+	switch (ADC_sensor_x) {
```

Comment:
This swtich statement isn't necessary. You could set Read_Position to ADC_sensor_x and ADC_count to 0 and then start ADC_SoftwareStartConv(ADC1). To do this, you'd either have to make the enum start at 0 or do (ADC_sensor_x - 1).